torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -495,6 +521,23 @@ std::vector<NCCLTraceBuffer::Entry> NCCLTraceBuffer::dump_entries() {
   return result;
 }
 
+// Returns the entry with the given id, if it exists. Otherwise, returns
+// std::nullopt.
+std::optional<NCCLTraceBuffer::Entry> NCCLTraceBuffer::getEntry(
+    std::optional<size_t> id) {
```

Comment:
its a little odd to let 'id' be an optional param to getEntry which then returns nullopt  if id is nullopt.

Reply:
Yeah, I went back and forth too. Eventually I decided to stay in line with existing API signatures, like:
`std::optional<size_t> NCCLTraceBuffer::record(...)`
`void NCCLTraceBuffer::retire_id(std::optional<size_t> id, ...)`